docs/source/api/notebooks/Notebook_3_Model_Variants.ipynb

```diff
@@ -0,0 +1,379 @@
+    "        self.set_gnn_encoder(gnn_encoder)\n",
+    "\n",
+    "        # set a decoder specific to node classification task\n",
+    "        decoder = EntityClassifier(in_dim=hid_size, num_classes=num_classes, multilabel=multilabel)\n",
+    "        self.set_decoder(decoder)\n",
+    "\n",
+    "        # classification loss function\n",
+    "        self.set_loss_func(ClassifyLossFunc(multilabel=multilabel))\n",
+    "\n",
+    "        # initialize model's optimizer\n",
+    "        self.init_optimizer(lr=0.001, sparse_optimizer_lr=0.01, weight_decay=0)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.1 GraphStorm built-in model architecture\n",
+    "\n",
+    "A GraphStorm built-in model normally contains four modules:\n",
+    "\n",
+    "- An input encoder that converts input node features to the embeddings with hidden dimensions.\n",
+    "- A GNN encoder that takes the embeddings from the input layer and performs message passing computation.\n",
+    "- A decoder that is task sepcific, e.g., the `EntityClassifier` for classification tasks.\n",
+    "- A loss function that matches specific tasks, e.g., the `ClassifyLossFunc`.\n",
+    "\n",
+    "Besides the four modules, a GraphStorm GNN model also need to initialize its own optimizer object."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.2 Model arguments\n",
+    "\n",
+    "Each specific GNN model may has its own model arguments. Some arguments could be common for other models, like the input and output dimensions, while others may be model specific. For example, `RGCN` model asks for the number of bases to reduce the number of learnable parameters, and attention-based models may need set the number of attention heads. Not only GNN models ask for arguments, GML tasks need specific arguments. For example, classification tasks may have multiple labels.\n",
+    "\n",
+    "GraphStorm APIs have given default values to many arguments. For better flexibility, we can add some arguments into model initialization, such as `num_hid_layers` and `hid_size`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.3 GML task modules\n",
+    "\n",
+    "Besides model-related modules, a GNN model also contains task-specific modules, including task specific decoders and loss functions. For example, to perform node classification task, the above `RgcnNCModel` model chooses the `EntityClassifier` as decoder and use the `ClassifyLossFunc` as its loss function."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "----\n",
+    "## 2 Reproduce GraphStorm Built-in GNN Model Variants\n",
+    "\n",
+    "With knowing the common architecture and arguments, it is easy to reproduce GraphStorm built-in GNN model variants.\n",
+    "\n",
+    "### 2.1 Reproduce an `RGAT` Model for Node Classification\n",
```

Comment:
Reproduce -> implement?

Reply:
Same as the above comment. `Reproduce` is more accurate than `Implement`.